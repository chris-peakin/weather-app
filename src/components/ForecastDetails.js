import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import "../styles/ForecastDetails.css";

function ForecastDetails({ details }) {
  const { date, temperature, humidity, wind } = details;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__temperaturemax">
        Max: {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__temperaturemin">
        Min: {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity} %</div>
      <div className="forecast-details__windspeed">
        Wind Speed: {wind.speed} mph
      </div>
      <div className="forecast-details__winddirection">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  details: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
