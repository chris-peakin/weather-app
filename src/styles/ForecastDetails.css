.forecast-details{
    text-align: center;
    font-weight: bold;
}