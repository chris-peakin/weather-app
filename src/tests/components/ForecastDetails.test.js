import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
        details: {
            date: 111111,
            temperature: {
                max: 11,
                min: 4,
            },
            humidity: 30,
            wind: {
                speed: 13,
                direction: "s",
            },
    },
};
    it("renders correctly", () => {
    const { asFragment } = render(
            <ForecastDetails
                details={validProps.details}
                />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});