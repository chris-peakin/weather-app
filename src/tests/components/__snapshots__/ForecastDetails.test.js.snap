// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details__temperaturemax"
    >
      Max: 11°C
    </div>
    <div
      class="forecast-details__temperaturemin"
    >
      Min: 4°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30 %
    </div>
    <div
      class="forecast-details__windspeed"
    >
      Wind Speed: 13 mph
    </div>
    <div
      class="forecast-details__winddirection"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
